DllImportGenerator/DllImportGenerator/Marshalling/BlittableArrayMarshaller.cs

```diff
@@ -0,0 +1,238 @@
+                                                                            IdentifierName(managedIdentifer),
+                                                                            IdentifierName("Length")))}))),
+                                            IdentifierName("CopyTo")))
+                                    .WithArgumentList(
+                                        ArgumentList(
+                                            SingletonSeparatedList(
+                                                Argument(IdentifierName(managedIdentifer))))))),
+                            ElseClause(
+                                ExpressionStatement(AssignmentExpression(SyntaxKind.SimpleAssignmentExpression,
+                                    IdentifierName(managedIdentifer),
+                                    LiteralExpression(SyntaxKind.NullLiteralExpression)))));
+                    }
+                    break;
+                case StubCodeContext.Stage.Cleanup:
+                    yield return GenerateConditionalAllocationFreeSyntax(info, context);
+                    break;
+            }
+        }
+
+        public override bool UsesNativeIdentifier(TypePositionInfo info, StubCodeContext context)
+        {
+            return (info.IsByRef || info.IsManagedReturnPosition) || !context.PinningSupported;
+        }
+
+        protected override ExpressionSyntax GenerateAllocationExpression(TypePositionInfo info, StubCodeContext context, SyntaxToken byteLengthIdentifier, out bool allocationRequiresByteLength)
+        {
+            allocationRequiresByteLength = true;
+            // (<nativeType>)Marshal.AllocCoTaskMem(<byteLengthIdentifier>)
+            return CastExpression(AsNativeType(info),
+                InvocationExpression(
+                    MemberAccessExpression(SyntaxKind.SimpleMemberAccessExpression,
+                        ParseTypeName(TypeNames.System_Runtime_InteropServices_Marshal),
+                        IdentifierName("AllocCoTaskMem")),
+                    ArgumentList(SingletonSeparatedList(Argument(IdentifierName(byteLengthIdentifier))))));
+        }
+
+        protected override ExpressionSyntax GenerateByteLengthCalculationExpression(TypePositionInfo info, StubCodeContext context)
+        {
+            // sizeof(<nativeElementType>) * <managedIdentifier>.Length
+            return BinaryExpression(SyntaxKind.MultiplyExpression,
+                SizeOfExpression(GetElementTypeSyntax(info)),
+                MemberAccessExpression(SyntaxKind.SimpleMemberAccessExpression,
+                    IdentifierName(context.GetIdentifiers(info).managed),
+                    IdentifierName("Length")
+                ));
+        }
+
+        protected override StatementSyntax GenerateStackallocOnlyValueMarshalling(TypePositionInfo info, StubCodeContext context, SyntaxToken byteLengthIdentifier, SyntaxToken stackAllocPtrIdentifier)
+        {
+            return EmptyStatement();
```

Comment:
Will this actually show up as an empty statement (`;`) in the generated code?

Reply:
Yes it will.